std/algorithm/searching.d

```diff
@@ -1080,40 +1080,38 @@ if (isBidirectionalRange!Range && Needles.length > 1 &&
 
 /// Ditto
 bool endsWith(alias pred = "a == b", R1, R2)(R1 doesThisEnd, R2 withThis)
-if (isBidirectionalRange!R1 &&
-    isBidirectionalRange!R2 &&
-    is(typeof(binaryFun!pred(doesThisEnd.back, withThis.back)) : bool))
+if (isBidirectionalRange!R1 && (
+        is(typeof(binaryFun!pred(doesThisEnd.back, withThis)) : bool) ||
+        isBidirectionalRange!R2 && (is(typeof(binaryFun!pred(doesThisEnd.back, withThis.back)) : bool))
+    ))
```

Comment:
The next overload is `isBidirectionalRange!R && is(typeof(binaryFun!pred(doesThisEnd.back, withThis)) : bool))`
-> `isBidirectionalRange!R` is the overlapping constraint.